src/Microsoft.Framework.Runtime/Internal/ConcurrencyUtilities.cs

```diff
@@ -19,7 +19,7 @@ internal static string FilePathToLockName(string filePath)
             return filePath.Replace(Path.DirectorySeparatorChar, '_');
         }
 
-        internal async static Task<T> ExecuteWithFileLocked<T>(string filePath, Func<bool, Task<T>> action)
+        public async static Task<T> ExecuteWithFileLocked<T>(string filePath, Func<bool, Task<T>> action)
```

Comment:
Could we have a synchronous overload for this which takes an `Action`? We can have it call this method and perform `GetAwaiter().GetResult()` in that method. @ChengTian is using this in another PR to perform synchronous operations and it would be cleaner to commonize the pattern.